### ISSUE TITLE
crash parsing a sentence

### ISSUE THREAD

**Issue by agayardo** [opencog/link-grammar#57]
Parsing the following text causes a segmentation fault in the parser: 
```
92 Chapter 10.............................................................................................................................................
```

This is reproducible both on Windows and in the Linux command-line link-parser, and this happens every time when we try to parse a document containing a table of contents. We will amend our sentence splitting and filtering to work this around, but perhaps this is worth looking at on the parser side too.

**Comment by ampli** [comment]
I have just sent a pull request that fixes it (#58).
It prevents this crash, but it doesn't split the page numbers and the dots.
This is because the code infrastructure (separate_word() and strip_right()) doesn't support such a separation for now.

(BTW, my wordgraph version (in my wordgraph1 branch) makes such a separation, but it also tries to split the long periods sequence in all possible ways, a thing that causes a long parsing time and an unneedly large number of linkages. I will try to fix it too.)

**Comment by agayardo** [comment]
Thanks a lot!
